katz/src/main/kotlin/katz/typeclasses/Applicative.kt

```diff
@@ -42,31 +41,257 @@ data class Tuple5<out A, out B, out C, out D, out E>(val a: A, val b: B, val c:
 data class Tuple6<out A, out B, out C, out D, out E, out F>(val a: A, val b: B, val c: C, val d: D, val e: E, val f: F)
 data class Tuple7<out A, out B, out C, out D, out E, out F, out G>(val a: A, val b: B, val c: C, val d: D, val e: E, val f: F, val g: G)
 data class Tuple8<out A, out B, out C, out D, out E, out F, out G, out H>(val a: A, val b: B, val c: C, val d: D, val e: E, val f: F, val g: G, val h: H)
+data class Tuple9<out A, out B, out C, out D, out E, out F, out G, out H, out I>(val a: A, val b: B, val c: C, val d: D, val e: E, val f: F, val g: G, val h: H, val i: I)
+data class Tuple10<out A, out B, out C, out D, out E, out F, out G, out H, out I, out J>(val a: A, val b: B, val c: C, val d: D, val e: E, val f: F, val g: G, val h: H, val i: I, val j: J)
 
 
-fun <F, A, B> HK<F, A>.product(AP: Applicative<F>, other: HK<F, B>): HK<F, Tuple2<A, B>> =
+fun <μ, A, Z> HK<μ, A>.product(AP: Applicative<μ>, other: HK<μ, Z>): HK<μ, Tuple2<A, Z>> =
```

Comment:
I'd avoid using μ, because the Android JVM isn't too happy about non-regular characters.